README.txt
To run the project:
step 2: npm i
step 2: npx expo start

CHANGES.md
# Changelog

## 0.1
- Initial release
- added useState to store the title and description value
- Added TextInput to show the text and edit the text instead of Text component
- Added new library react-native-responsive-screen
- changes the button style
- 

App.js
import React, { useState } from "react";
import {
  StyleSheet,
  ScrollView,
  ImageBackground,
  Dimensions,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import Phase from "./components/Phase";
import phasesData from "./data/phases.json";
import Button from "./components/Button";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";

const screenHeight = Dimensions.get("window").height;
const screenWidth = Dimensions.get("window").width;

const App = () => {
  const [phases, setPhases] = useState(phasesData);

  // ADD EACH PHASE
  const addPhase = () => {
    const newPhase = { id: phases.length + 1 };
    setPhases((prev) => [...prev, newPhase]);
  };

  // REMOVE EACH PHASE
  const deletePhase = (id) => {
    const filteredPhases = phases.filter((phase) => phase.id !== id);
    setPhases(filteredPhases);
  };

  return (
    <ImageBackground
      source={require("./assets/background.jpg")}
      style={styles.backgroundImage}
      resizeMode="stretch"
    >
      <Text style={styles.title}>MW - TODO</Text>
      <ScrollView horizontal style={styles.container}>
        {phases.map((phase, index) => (
          <View key={index}>
            <TouchableOpacity onPress={() => deletePhase(phase.id)}>
              <Text>Delete Phase</Text>
            </TouchableOpacity>
            <Phase key={phase.id} phase={phase} />
          </View>
        ))}

        <TouchableOpacity style={styles.buttonContainer} onPress={addPhase}>
          <Text style={styles.buttonText}>+ Add phase</Text>
        </TouchableOpacity>
      </ScrollView>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  title: {
    fontSize: 24,
    color: "#fff",
    fontWeight: "bold",
    marginTop: 48,
    marginBottom: 6,
  },
  container: {
    margin: 5,
    marginRight: 16,
  },
  backgroundImage: {
    height: screenHeight,
    width: screenWidth,
    justifyContent: "center",
    alignItems: "center",
  },
  buttonContainer: {
    backgroundColor: "#000",
    height: hp(5),
    width: wp(25),
    borderRadius: 5,
  },
  buttonText: {
    color: "#fff",
    textAlign: "center",
    marginTop: 5,
  },
});
export default App;


components/Card.js
import React, { useState } from "react";
import {
  View,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  Text,
} from "react-native";
import Draggable from "react-native-draggable";
const Card = ({ card }) => {
  const [data, setData] = useState(card);
  const [title, setTitle] = useState(data.title);
  const [desc, setDesc] = useState(data.description);

  const titleChange = (text) => {
    setTitle(text);
  };

  const descChange = (text) => {
    setDesc(text);
  };

  return (
    <View style={styles.container}>
      <View>
        <TextInput
          style={styles.title}
          onChangeText={(text) => titleChange(text)}
          value={title}
        />

        <TextInput
          onChangeText={(text) => descChange(text)}
          style={styles.description}
          value={desc}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 10,
    borderWidth: 1,
    borderRadius: 5,
    marginVertical: 5,
    backgroundColor: "#ffffff",
    width: "100%",
  },
  title: {
    fontSize: 16,
  },
  description: {
    fontSize: 12,
  },
  draggable: {
    zIndex: 999,
  },
});

export default Card;


components/Phase.js
import React, { useState } from "react";
import {
  View,
  Text,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  Dimensions,
} from "react-native";
import Card from "./Card";
import Button from "./Button";
import { Button as RnButton } from "react-native";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";
import Draggable from "react-native-draggable";

const screenHeight = Dimensions.get("window").height;
const screenWidth = Dimensions.get("window").width;
const Phase = ({ phase }) => {
  const [cards, setCards] = useState(phase.cards || []);
  const [index, setIndex] = useState(0);

  // ADD CARD
  const addCard = () => {
    const newCard = { id: index };
    setCards((prevCards) => [...prevCards, newCard]);
    setIndex(index + 1);
  };

  //REMOVE EACH CARD
  const deleteCard = (id) => {
    const filteredCards = cards.filter((card) => card.id !== id);
    setCards(filteredCards);
  };

  return (
    <View>
      <View style={styles.container}>
        <View style={styles.header}>
          <Text style={styles.title}>{phase.title}</Text>
          <RnButton title="..." color={"#fff"} />
        </View>

        <ScrollView>
          {cards.map((card, index) => (
            <View key={index}>
              <TouchableOpacity onPress={() => deleteCard(card.id)}>
                <Text>Delete Card</Text>
              </TouchableOpacity>
              <Card key={index} card={card} />
            </View>
          ))}
          <TouchableOpacity style={styles.buttonContainer} onPress={addCard}>
            <Text style={styles.buttonText}>+ Add card</Text>
          </TouchableOpacity>
        </ScrollView>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: screenHeight,
    width: screenWidth,
    padding: 10,
    backgroundColor: "rgba(0, 0, 0, 0.5)",
    borderRadius: 10,
    marginRight: 10,
    marginBottom: 30,
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    paddingHorizontal: 8,
    paddingBottom: 8,
  },
  title: {
    fontSize: 16,
    fontWeight: "bold",
    color: "#fff",
  },
  option: {
    fontSize: 24,
    fontWeight: "bold",
    color: "#fff",
  },
  buttonContainer: {
    backgroundColor: "#000",
    height: hp(5),
    width: wp(25),
    borderRadius: 5,
  },
  buttonText: {
    color: "#fff",
    textAlign: "center",
    marginTop: 5,
  },
});

export default Phase;
